api/v1alpha1/lvmvolumegroup_webhook.go

```diff
@@ -0,0 +1,85 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// log is for logging in this package.
+var lvmvolumegrouplog = logf.Log.WithName("lvmvolumegroup-webhook")
+
+var _ webhook.Validator = &LVMVolumeGroup{}
+
+//+kubebuilder:webhook:path=/validate-lvm-topolvm-io-v1alpha1-lvmvolumegroup,mutating=false,failurePolicy=fail,sideEffects=None,groups=lvm.topolvm.io,resources=lvmvolumegroups,verbs=create;update,versions=v1alpha1,name=vlvmvolumegroup.kb.io,admissionReviewVersions=v1
+
+func (in *LVMVolumeGroup) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(in).
+		Complete()
+}
+
+func (in *LVMVolumeGroup) ValidateCreate() (warnings admission.Warnings, err error) {
+	lvmvolumegrouplog.Info("validate create", "name", in.Name)
+	return in.ValidateUpdate(nil)
+}
+
+func (in *LVMVolumeGroup) ValidateUpdate(_ runtime.Object) (warnings admission.Warnings, err error) {
+	lvmvolumegrouplog.Info("validate update", "name", in.Name)
+	if err := in.validateDeviceSelector(); err != nil {
+		return nil, fmt.Errorf(".Spec.DeviceSelector is invalid: %w", err)
+	}
+	return nil, nil
+}
+
+func (in *LVMVolumeGroup) validateDeviceSelector() error {
```

Comment:
Why don't we validate this in LVMCluster webhook?

Reply:
We do actually, just with more detailed errors: `verifyNoDeviceOverlap` in the LVMCluster webhook contains a similar logic to `validateDeviceSelector`. I think we can also decide to remove this validation completely, but then it would be possibly to create an LVMVolumeGroup externally that skips this validation. Alternatively we can also unify the logic of both paths, however then I have to change the error reporting at one of the two places since they report duplicates differently. What's your preference here?

Reply:
As `LVMVolumeGroup` is an internally used CRD, I'd vote for removing it for simplicity. If someone creates an `LVMVolumeGroup` CR externally, it might lead to bigger problems than duplicate device paths anyway.